packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/ImageAndTextViewer/ImageAndTextViewerContainer.js

```diff
@@ -0,0 +1,202 @@
+  frame = 0,
+  ImageObject = window.Image,
+  loadingState = asyncStates.initialized,
+  move,
+  onError = () => true,
+  onKeyDown = () => true,
+  onReady = () => true,
+  rotation,
+  setFrame = () => true,
+  setOnPan = () => true,
+  setOnZoom = () => true,
+  subject
+}) {
+  const subjectImage = useRef()
+  const [dragMove, setDragMove] = useState()
+
+  // TODO: refactor with frame
+  const [location, setLocation] = useState(0)
+
+  // TODO: replace this with a better function to parse the image location from a subject.
+  const imageUrl = subject ? Object.values(subject.locations[frame])[0] : null
+  const { img, error } = useSubjectImage(ImageObject, imageUrl)
+  // default to a placeholder while image is loading.
+  const { naturalHeight, naturalWidth, src } = img
+
+  useEffect(function onImageLoad () {
+    if (src !== placeholder.src) {
+      const svgImage = subjectImage.current
+      const { width: clientWidth, height: clientHeight } = svgImage
+        ? svgImage.getBoundingClientRect()
+        : {}
+      const target = { clientHeight, clientWidth, naturalHeight, naturalWidth }
+      onReady({ target })
+    }
+  }, [src])
+
+  useEffect(function onMount () {
+    enableRotation()
+  }, [])
+
+  useEffect(function onFrameChange () {
+    activeTool?.validate()
+  }, [frame])
+
+  if (error) {
+    console.error(error)
+    onError(error)
+  }
+
+  function setOnDrag (callback) {
+    setDragMove(() => callback)
+  }
+
+  function onDrag (event, difference) {
+    dragMove?.(event, difference)
+  }
+
+  function onLocationChange () {
+    setLocation(Math.abs(location - 1))
+  }
+
+  if (loadingState === asyncStates.error) {
+    return (
+      <div>Something went wrong.</div>
+    )
+  }
+
+  const enableDrawing = (loadingState === asyncStates.success) && enableInteractionLayer
+  const SubjectImage = move ? DraggableImage : 'image'
+  const subjectImageProps = {
+    height: naturalHeight,
+    width: naturalWidth,
+    xlinkHref: src,
+    ...(move && { dragMove: onDrag })
+  }
+
+  if (loadingState !== asyncStates.initialized) {
+    return (
+      <Box
+        fill='horizontal'
+      >
+        <SVGPanZoom
+          img={subjectImage.current}
+          maxZoom={5}
+          minZoom={0.1}
+          naturalHeight={naturalHeight}
+          naturalWidth={naturalWidth}
+          setOnDrag={setOnDrag}
+          setOnPan={setOnPan}
+          setOnZoom={setOnZoom}
+          src={src}
+        >
+          {location === 0 ? (
+            <SingleImageViewer
+              enableInteractionLayer={enableDrawing}
+              height={naturalHeight}
+              onKeyDown={onKeyDown}
+              rotate={rotation}
+              width={naturalWidth}
+            >
+              <g ref={subjectImage}>
+                <SubjectImage
+                  {...subjectImageProps}
+                />
+              </g>
+            </SingleImageViewer>
+          ) : (
+            <SingleTextViewer />
```

Comment:
You're rendering the text viewer inside an `<svg>` element here, so you'll have to either rewrite it in SVG, or [embed HTML in SVG as an object](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/foreignObject).

Reply:
Refactored to check location type and render appropriate viewer.